gitclone/checkout.go

```diff
@@ -54,6 +54,11 @@ func NewParameterValidationError(msg string) error {
 // checkoutStrategy is the interface an actual checkout strategy implements
 type checkoutStrategy interface {
 	do(gitCmd git.Git, fetchOptions fetchOptions, fallback fallbackRetry) error
+	// getBuildTriggerRef returns ref to the commit/branch/tag that triggered the build.
+	// For simple checkout strategies the returned ref will be HEAD (after running 'do').
+	// However a PR checkout strategy may create a (temporary) merge commit, so the merged state can be tested.
+	// In this case the returned ref will point to the Source branch (or a commit on the Source branch).
```

Comment:
Would it make sense to mention manually triggered builds here?

Reply:
From the Step perspective it does not matter if it is manually trigerred or trigerred by webhook. It can also run locally (in theory at least).
So I am unsure why that would help.

Reply:
Makes sense. I didn't know, so that's why I asked.

Reply:
BTW Part of the complexity in the Step is caused by the fact that the webhhook handler does not send ready made checkout strategies, but a bunch of envs.